client/src/pages/Messenger.js

```diff
@@ -1,33 +1,26 @@
-import React, { useState, useRef, useEffect } from "react";
+import React, { useState, useEffect } from "react";
 import { useHistory } from "react-router-dom";
 import io from "socket.io-client";
 
 import Sidebar from "../components/sidebar/Sidebar";
 import Convo from "../components/convo/Convo";
 import "./Messenger.css";
-
-// Connection options to bypass CORS warning
-const connectionOptions = {
-	"force new connection": true,
-	reconnectionAttempts: "Infinity",
-	timeout: 10000,
-	transports: ["websocket"]
-};
+import {
+	connectionOptions,
+	socketServerURL
+} from "../utilities/socketConfiguration";
 
 // Initializing our socket based on dev or prod server
-let SOCKET_SERVER_URL;
-if (process.env.NODE_ENV === "production") {
-	SOCKET_SERVER_URL = "https://decipher-io.herokuapp.com/";
-} else {
-	SOCKET_SERVER_URL = "http://localhost:5000/";
-}
+let socket,
+	SOCKET_SERVER_URL = socketServerURL();
 
 const Messenger = () => {
 	// Your profile information
 	const [user, setUser] = useState({});
 	// Keeps track of your current conversation
 	const [currConvo, setCurrConvo] = useState(null);
-	const socketRef = useRef();
+	// Keeps track of whether message is sent and triggers re-render
+	const [isMessageSent, setIsMessageSent] = useState({ status: false });
```

Comment:
will you be extending this more in the future? if not, instead of an object, just store a boolean

Reply:
Nope, not at this moment. I just stored it as boolean!